client/client.go

```diff
@@ -40,29 +40,21 @@ type DataExportClient interface {
 type Client struct {
 	HTTPClient *http.Client
 	Config     *config.Config
+	AuthFunc   func(*http.Request)
 }
 
 var _ DataExportClient = (*Client)(nil)
 
-// NewClient returns a Client initialized with http.DefaultClient and the
-// supplied apiToken.
-func NewClient(config *config.Config) *Client {
-	return &Client{
-		HTTPClient: http.DefaultClient,
-		Config:     config,
-	}
+func (c *Client) AddAuthHeaders(req *http.Request) {
```

Comment:
nit: I would not expose anything like this.

Just allow caller to supply a custom HTTP client (not concrete type but an interface -- like `DataExportClient` or maybe just `http.RoundTripper`) and then have hauser's `main` create a client that automatically adds the configured API key as an auth header. That way, any sort of customization is just done via decorating the HTTP client.

Reply:
I don't know if it will suffice for what you wanted but:

* Made this function private (addAuthHeaders)
* Allowed the auth func to be passed into the constructor of Client()
* Made all the fields of Client private
* Put NewClient() back to client package and removed additional parameters from the default path used by Hauser's main.  Thus Hauser's main is now unchanged and there is the _option_ to supply an auth func, that Hauser's main does not use.

This minimizes duplication and the only thing is exposed is the option to supply your own auth func at construction time. Everything else is now private.

@camphillips22  ^^^